src/EventEmitter.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * A generic handler for incoming event data.
+ */
+type THandler<T> = (data: T) => void;
+
+/**
+ * Simple object for keeping track of event listeners.
+ */
+interface Events<T> {
+  [index: string]: Array<THandler<T>>;
+}
+
+const findIndex =
+  [].findIndex ||
```

Comment:
I know we said earlier that `Set` wasn't an option for IE11, but the [MDN compatiability table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#Browser_compatibility) makes it seem like it's compatible enough.